static/app/views/onboarding/targetedOnboarding/components/stepper.tsx

```diff
@@ -0,0 +1,76 @@
+import {HTMLAttributes, useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+import {animate, motion, useMotionValue} from 'framer-motion';
+
+import {StepDescriptor} from '../types';
+
+const StepperWrapper = styled('div')`
+  border-radius: 4px;
+  position: relative;
+  overflow: hidden;
+`;
+const StepperContainer = styled('div')`
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+`;
+
+const StepperIndicator = styled('span')<{clickable?: boolean}>`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBackground};
+  cursor: ${p => (p.clickable ? 'pointer' : 'default')};
+`;
+
+const StepperTransitionIndicator = styled(motion.span)`
+  height: 8px;
+  width: 80px;
+  background-color: ${p => p.theme.progressBar};
+  position: absolute;
+`;
+
+type Props = Omit<HTMLAttributes<HTMLDivElement>, 'onClick'> & {
+  currentStepId: string;
+  onClick: (step: StepDescriptor) => void;
+  steps: StepDescriptor[];
```

Comment:
@Neo-Zhixing I feel like we don't need to expose this datatype to this component. Can't we just pass in number of steps and use the step index? That would make it easier to use this component outside of onboarding if we wanted. We're not really using any of the fields of the `StepDescriptor` except in determining the index as far as I can tell.

Reply:
Good call. Will change